src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionExecutor.cs

```diff
@@ -58,7 +59,7 @@ public static class ControllerActionExecutor
                         if (defaultValueAttribute?.Value == null)
                         {
                             value = parameterInfo.ParameterType.GetTypeInfo().IsValueType
-                                ? Activator.CreateInstance(parameterInfo.ParameterType)
+                                ? actionMethodExecutor.CreateParameterObject(index)
```

Comment:
A better and simpler way to do this would probably be to add support to `ObjectMethodExecutor` for storing a 'default value' per parameter. We're just dealing with value types here so it should be safe. 

We should also include the result of the call to `MethodInfo.GetParameters()` which allocates.

Notice that this code is looking for `DefaultValueAttribute`, we should put that check up front as well.